src/commands/staff/ShowBlackList.ts

```diff
@@ -0,0 +1,60 @@
+import { BaseCommand, ICommandContext, ICommandInfo } from "../BaseCommand";
+import { MessageAttachment, } from "discord.js";
+import { StringBuilder } from "../../utilities/StringBuilder";
+
+
+export class ShowBlacklist extends BaseCommand {
+    public constructor() {
+        const cmi: ICommandInfo = {
+            cmdCode: "SHOW_BLACKLIST",
+            formalCommandName: "Show Blacklist",
+            botCommandName: "showblacklist",
+            description: "Shows a list of all current blacklisted users of the server",
+            rolePermissions: ["Officer", "Moderator", "HeadRaidLeader", "Security"],
+            generalPermissions: [],
+            botPermissions: [],
+            commandCooldown: 3 * 1000,
+            argumentInfo: [],
+            guildOnly: true,
+            botOwnerOnly: false
+        };
+
+        super(cmi);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    //add command to index of commands and bot.ts
+    public async run(ctx: ICommandContext): Promise<number> {
+        const blInfo = ctx.guildDoc!.moderation.blacklistedUsers.map(x => ` Realm Name: ${x.realmName.lowercaseIgn}` + 
+        ` - Reason: ${x.reason}`).join("\n");
+
+        const usersBf = new StringBuilder()
+            .append("================= BlackListed Users =================").appendLine()
+            .append(blInfo).appendLine()
+            .append("======================= END =========================").appendLine()
+            .toString();
+
+        if (!blInfo) {
```

Comment:
If there are no blacklisted members (I.e., `blInfo.length` is 0), then this will run instead of the else statement right below it. So, this condition is unnecessary.

Reply:
<img width="478" alt="Screen Shot 2022-06-30 at 5 24 10 PM" src="https://user-images.githubusercontent.com/78934157/176788701-6a6cdf75-3667-48b8-aab3-58048805c2a4.png">

Because array is of type object an empty array evaluates to true. ([] == false is right) but not ====

one checks if it is falsely the other checks if the array is empty.